### PR TITLE
Add style guide entries for strict data and Pretty

### PR DESCRIPTION
**Pull request by michaelpj**

### CODE REVIEW

File: STYLEGUIDE.adoc [hunk 1]
```diff
@@ -240,6 +240,23 @@ Use Text instead of String unless you have a good reason not to.
 
 It’s 2022, use a proper, unicode aware string type instead of a linked list.
 
+=== Derive `Show`, define `Pretty`, use `Pretty` for human-readable output
+
+Always derive `Show`, do not define it manually.
```
Comment (effectfully) [inline STYLEGUIDE.adoc:245]: What about exception types? `Show` is hardcoded into `Exception`, but we'd like exception messages to be readable.

File: STYLEGUIDE.adoc [hunk 2]
```diff
@@ -280,6 +297,19 @@ Avoid orphan instances, but don’t worry about it if https://www.michaelpj.com/
 
 See the blog post.
 
+=== Make data strict by default, use `nothunks` for long-lived data structures
+
+Use `StrictData` for new code; make fields strict unless you have a good reason not to.
```
Comment (effectfully) [inline STYLEGUIDE.adoc:302]: :tada:

**Review by effectfully — APPROVED** [review]

File: STYLEGUIDE.adoc (hunk 1 above)
Comment (michaelpj) [reply STYLEGUIDE.adoc:245]: I thought you could just define `displayException`? https://hackage.haskell.org/package/base-4.16.1.0/docs/Control-Exception.html#v:displayException

Comment (michaelpj) [reply STYLEGUIDE.adoc:245]: Worth including, though.

**Comment by michaelpj** [comment]
test

**Comment by michaelpj** [comment]
test

File: STYLEGUIDE.adoc [hunk 3]
```diff
@@ -240,6 +240,23 @@ Use Text instead of String unless you have a good reason not to.
...
+
+Always derive `Show`, do not define it manually.
+
+Always use `Pretty` for human-readable output, not `Show`.
+
+Always define `Pretty` explicitly (when you need it). It's okay to delegate to the `Show` instance if you think it's good enough.
+
+==== Reasoning
+
+The derived version of `Show` is always useful as a way of seeing the explicit strucuture of a value as a Haskell value.
```
Comment (thealmarty) [inline STYLEGUIDE.adoc:253]:
```suggestion
The derived version of `Show` is always useful as a way of seeing the explicit structure of a value as a Haskell value.
```

**Review by thealmarty — APPROVED** [review]
LGTM! Maybe change the line limit to 100 along the way?

**Comment by michaelpj** [comment]
> Maybe change the line limit to 100 along the way?

I was actually going to do that separately, I have a thing for it.

File: STYLEGUIDE.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (michaelpj) [reply STYLEGUIDE.adoc:302]: I wanted to imply that people shouldn't blindly add it where it isn't already, but maybe it's clear enough to say it unconditionally. All of this stuff is guidelines anyway.

**Pull request merged**